src/transformers/models/ctrl/modeling_ctrl.py

```diff
@@ -359,18 +359,18 @@ def _prune_heads(self, heads_to_prune):
     )
     def forward(
         self,
-        input_ids=None,
-        past_key_values=None,
-        attention_mask=None,
-        token_type_ids=None,
-        position_ids=None,
-        head_mask=None,
-        inputs_embeds=None,
-        use_cache=None,
-        output_attentions=None,
-        output_hidden_states=None,
-        return_dict=None,
-    ):
+        input_ids: torch.LongTensor = None,
+        past_key_values: Tuple[Tuple[torch.FloatTensor]] = None,
```

Comment:
Those two are optional as well (this applies to all the other forward pass you touched).

Reply:
@sgugger thanks for your comment. Incorporated.